### ISSUE TITLE
Clippy raising `undefined symbol` on nightly 76242aebb 2017-06-06

### ISSUE THREAD

**Issue by snim2** [Manishearth/rust-clippy#1817]
Rust version:

```sh
$ ~/bin/rustup.sh --channel=nightly
rustup: gpg available. signatures will be verified
rustup: downloading manifest for 'nightly'
rustup: downloading toolchain for 'nightly'
######################################################################## 100.0%
gpg: assuming signed data in '~/.rustup/dl/72ed7bb8a6711cba6803/rust-nightly-x86_64-unknown-linux-gnu.tar.gz'
gpg: Signature made Wed 07 Jun 2017 01:06:42 BST
gpg:                using RSA key 5CB4A9347B3B09DC
gpg: Good signature from "Rust Language (Tag and Release Signing Key) <rust-key@rust-lang.org>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: 108F 6620 5EAE B0AA A8DD  5E1C 85AB 96E6 FA1B E5FE
     Subkey fingerprint: C134 66B7 E169 A085 1886  3216 5CB4 A934 7B3B 09DC
rustup: installing toolchain for 'nightly'

...
    Rust is ready to roll.

$ rustc --version
rustc 1.19.0-nightly (76242aebb 2017-06-06)
$
```

Clippy behaviour:

```sh
$ cargo-clippy --version
cargo-clippy: symbol lookup error: cargo-clippy: undefined symbol: _ZN68_$LT$rustc_const_math..int..ConstInt$u20$as$u20$core..ops..BitOr$GT$5bitor17h5c3a6e170c3c4c9eE

$ cargo clippy # On a random project
~/.cargo/bin/cargo-clippy: symbol lookup error: ~/.cargo/bin/cargo-clippy: undefined symbol: _ZN68_$LT$rustc_const_math..int..ConstInt$u20$as$u20$core..ops..BitOr$GT$5bitor17h5c3a6e170c3c4c9eE
```

**Comment by mcarton** [comment]
You need to reinstall Clippy everytime you update rustc.

**Comment by snim2** [comment]
Sorry, I realise now that the issue report wasn't very clear on this matter. I tried uninstalling and re-installing clippy with Cargo, I also tried nuking my `~/.cargo/` directory and starting again. Neither worked.

**Comment by mcarton** [comment]
Try with `rustup run nightly cargo-clippy` or had the directory with rustc's libs to your path.
You might also want to update to [`rustup.rs`](https://github.com/rust-lang-nursery/rustup.rs), [`rustup.sh`](https://github.com/rust-lang/rustup.sh) is deprecated.

**Comment by oli-obk** [comment]
@snim2 any concrete reason you are using the old rustup sh script? I don't think it's supported anymore. It might be the reason for the failure, cargo clippy was intended to work with the rustup env vars, and I don't know what the old script configures.

You can still use it, but you'll need to check the env vars mentioned on the clippy readme.

**Comment by snim2** [comment]
Well, this explains it, I hand't realised that `rustup.sh` had been deprecated, clippy seems fine after moving to the new installer. Thanks for the help!
